### PR TITLE
Convert todt to d

### PR DESCRIPTION
**Pull request by yebblies**
@WalterBright

### CODE REVIEW

**Comment by yebblies** [comment]
Now actually working!

**Comment by yebblies** [comment]
Or maybe this time.

**Comment by yebblies** [comment]
@WalterBright It's passing now!

**Comment by 9rnsr** [comment]
Auto-merge toggled on

**Comment by yebblies** [comment]
Thanks!

File: src/backend/dt.h [hunk 1]
```diff
@@ -18,30 +18,36 @@ bool dtallzeros(const dt_t *dt);
...
 
     dt_t *head;
     dt_t **pTail;
 
-  public:
+public:
 
     DtBuilder();
-    dt_t *finish();
+    virtual dt_t *finish();
```
Comment (WalterBright) [inline src/backend/dt.h]: why is this virtual?

File: src/backend.d [hunk 2]
```diff
@@ -155,3 +155,43 @@ enum
...
+    dt_t** pTail;
+
+public:
+    this()
+    {
+        pTail = &head;
+    }
+
+extern (C++):
+    dt_t* finish();
```
Comment (WalterBright) [inline src/backend.d]: finish should be final, too

**Pull request merged**